release_automation.sh

```diff
@@ -287,15 +288,12 @@ execute "git" "push" "-u" "origin" "HEAD"
 ohai "Create release branch in WordPress-Android"
 execute "git" "switch" "-c" "$WP_APPS_INTEGRATION_BRANCH"
 
-ohai "Update gutenberg-mobile ref"
-cd libs/gutenberg-mobile
-execute "git" "fetch" "--recurse-submodules=no" "origin" "$GB_MOBILE_PR_REF"
-execute "git" "checkout" "$GB_MOBILE_PR_REF"
-execute "git" "submodule" "update"
-cd ../..
+ohai "Update build.gradle file with the latest version"
+test -f "build.gradle" || abort "Error: Could not find build.gradle"
+sed -i'.orig' -E "s/ext.gutenbergMobileVersion = '(.*)'/ext.gutenbergMobileVersion = 'v$VERSION_NUMBER'/" build.gradle || abort "Error: Failed updating gutenbergMobileVersion in build.gradle"
```

Comment:
I don't believe we can set the `vX.XX` version at this time because we're just cutting branches and creating PRs here, we're not actually creating the tagged release (which would result in the vX.XX binary). I think the workflow would go like this.

1. This script creates all the branches and updates the gb-mobile reference in WPAndroid to refer to the binary for the `HEAD` of the gb-mobile release PR ([for example](https://github.com/wordpress-mobile/WordPress-Android/pull/14386/commits/17e279ad89a1a50c578520ed2bf552de1ea43fbe))
2. Each time we update the gb-mobile release branch, we update the WPAndroid gb-mobile binary version to reflect the new `HEAD` of the release branch.
3. We "cut" the gb-mobile release, tagging it with the appropriate version CI picks up the tag and then builds the binary.
4. We manually update the WPAndroid branch to reflect the "tagged" gb-mobile version.

With that said, I haven't actually done this workflow yet myself, so definitely curious to hear any thoughts @jd-alexander  and @guarani  have.

Reply:
The points @mchowning laid out make sense to me. Because we tag the `gutenberg-mobile` release _after_ the script creates the `wordpress-android` branch, we cannot rely upon the version number. 

The proposed workflow steps also make sense. Using the tip of `HEAD` for the `gutenberg-mobile` release branch appears to be the approach that [Joel manually used in the previous release](https://github.com/wordpress-mobile/WordPress-Android/blob/17de6cf384659be96dbfa4d974666c5015da3b48/build.gradle#L10) via the `<PR_number>-<full_commit_sha>` pattern. So, if we are able to have the the script recreate this pattern, that likely would work. 

It looks like we "capture" the `gutenberg-mobile` [commit SHA](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/7c82b25d605525bacf6792dcb5b269f3ac2585b5/release_automation.sh#L250) and [PR URL](https://github.com/wordpress-mobile/release-toolkit-gutenberg-mobile/blob/7c82b25d605525bacf6792dcb5b269f3ac2585b5/release_automation.sh#L201) already. We may be able to parse the PR number from the latter. There doesn't appear to be [a `gh` CLI command](https://cli.github.com/manual/gh_pr) for requesting the PR number of the current branch. 

A short term fix could be temporarily removing the `Update gutenberg-mobile ref` section altogether. This would allow us to leverage the toolkit to create the branches and PR, but we'd manually update the `gutenbergMobileVersion`.

Reply:
I agree with the points mentioned above. Allowing the script to update the `gutenbergMobileVersion` with `<PR_number>-<full_commit_sha>` when the branch is created is the appropriate approach as reflected in the latest changes here. Once the gb-mobile release has been merged in we are then able to manually update the `gutenbergMobileVersion` with the gb-mobile release tag of the current release.